src/sema/types.rs

```diff
@@ -1490,7 +1490,7 @@ impl Type {
             Type::Ref(r) => r.is_reference_type(ns),
             Type::StorageRef(_, r) => r.is_reference_type(ns),
             Type::InternalFunction { .. } => false,
-            Type::ExternalFunction { .. } => false,
+            Type::ExternalFunction { .. } => true,
```

Comment:
On Ethereum, an external function fits into a 256 bit register, so it's not a reference type. Ideally this should be `ns.target != Target::EVM` with a comment.